hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -157,11 +157,17 @@ public Response getKeysForContainer(
       for (ContainerKeyPrefix containerKeyPrefix : containerKeyPrefixMap
           .keySet()) {
 
-        // Directly calling get() on the Key table instead of iterating since
-        // only full keys are supported now. When we change to using a prefix
-        // of the key, this needs to change to prefix seek.
-        OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable(getBucketLayout())
-            .getSkipCache(containerKeyPrefix.getKeyPrefix());
+      // Directly calling getSkipCache() on the Key/FileTable table accordingly
+      // instead of iterating since only full keys are supported now. We will
+      // try to get the OmKeyInfo object by searching the KEY_TABLE table with
+      // the key prefix. If it's not found, we will then search the FILE_TABLE
+        OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable(BucketLayout.LEGACY)
+            .getSkipCache(containerKeyPrefix.getKeyPrefix()) == null ?
+            omMetadataManager.getKeyTable(BucketLayout.FILE_SYSTEM_OPTIMIZED)
+                .getSkipCache(containerKeyPrefix.getKeyPrefix()) :
+            omMetadataManager.getKeyTable(BucketLayout.LEGACY)
+                .getSkipCache(containerKeyPrefix.getKeyPrefix());
```

Comment:
maybe we could avoid calling `omMetadataManager.getKeyTable(BucketLayout.LEGACY).getSkipCache(containerKeyPrefix.getKeyPrefix())` twice, either we could save the value to a variable and use that
```suggestion
        OmKeyInfo omKeyInfo = ((OmKeyInfo keyinfo =
        omMetadataManager.getKeyTable(BucketLayout.LEGACY).getSkipCache(
        containerKeyPrefix.getKeyPrefix())) == null) ?
        omMetadataManager.getKeyTable(BucketLayout.FILE_SYSTEM_OPTIMIZED)
        .getSkipCache(containerKeyPrefix.getKeyPrefix()) : keyinfo;
```
or we could do this
```suggestion
        OmKeyInfo omKeyInfo = Optional.ofNullable(omMetadataManager.getKeyTable(BucketLayout.LEGACY)
            .getSkipCache(containerKeyPrefix.getKeyPrefix()))
            .orElse(omMetadataManager.getKeyTable(BucketLayout.FILE_SYSTEM_OPTIMIZED)
                .getSkipCache(containerKeyPrefix.getKeyPrefix()));
```
let me know if I misunderstood something or you don't think this is necessary, but maybe it makes the code slightly more readable and we avoid a duplicated calling.

Reply:
Thanks for the suggestion @dombizita I have made the changes, do take a look !!

Reply:
thanks for addressing this, it looks good to me!